src/Raven.Server/ServerWide/Commands/Subscriptions/AcknowledgeSubscriptionBatchCommand.cs

```diff
@@ -60,10 +66,60 @@ protected override BlittableJsonReaderObject GetUpdatedValue(long index, RawData
                 return context.ReadObject(existingValue, SubscriptionName);
             }
 
-            if (LastKnownSubscriptionChangeVector != subscription.ChangeVectorForNextBatchStartingPoint)
-                throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't acknowledge subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscription.ChangeVectorForNextBatchStartingPoint}, received value: {LastKnownSubscriptionChangeVector}");
+            if (isSharded)
+            {
+                if (string.IsNullOrEmpty(LastKnownSubscriptionChangeVector))
+                {
+                    if (string.IsNullOrEmpty(subscription.ChangeVectorForNextBatchStartingPoint))
+                    {
+                        // can update
+                    }
+                    else
+                    {
+                        var currentLocalShardCv = ShardLocalChangeVector.ToChangeVector();
+                        var currentCvInStorage = subscription.ChangeVectorForNextBatchStartingPoint.ToChangeVector();
+
+                        foreach (var entry in currentLocalShardCv)
+                        {
+                            if (currentCvInStorage.Any(x => x.DbId == entry.DbId))
+                                throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't acknowledge sharded subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscription.ChangeVectorForNextBatchStartingPoint}, received value: {LastKnownSubscriptionChangeVector}");
+                        }
```

Comment:
what is the purpose of this check?

Reply:
Need an answer

Reply:
this is the case when the shard subscription ack's for first batch (LastKnownSubscriptionChangeVector is null), and there was ack of other shard meanwhile this shard was processing (subscription.ChangeVectorForNextBatchStartingPoint is not null) the batch, In such case I check that the current stored cv for this subscription (subscription.ChangeVectorForNextBatchStartingPoint) doesn't contain any part of shard's local change vector
which I pass to ack command, and assign while processign the batch
https://github.com/ravendb/ravendb/pull/12543/files#diff-e7d9590176a763b1ec6213d1f14ed32c69c74240497b13662394568dbef547bbR411